xl_auth/permission/forms.py

```diff
@@ -124,18 +129,16 @@ def validate_permission_id(self, field):
                                     permission_id=self.target_permission_id))
         current_collection = target_permission.collection
         form_collection = Collection.get_by_id(self.collection_id.data)
-        if not form_collection:
-            raise ValidationError(_('Collection ID "%(collection_id)s" does not exist',
-                                    collection_id=self.collection_id.data))
-
-        if not (self.current_user.is_cataloging_admin_for(
+        if form_collection and not (self.current_user.is_cataloging_admin_for(
```

Comment:
What about the case where `form_collection` is `None`? The way I read this is we'll validate `permission_id` if that's the case, which is a change from before.

Reply:
It actually duplicated the validator below, resulting in duplicate (and contradictory) error messages when the user failed to select a collection.

This one below is what makes *form_collection* cannot be none without validation failure: 

```python

    # noinspection PyMethodMayBeStatic
    def validate_collection_id(self, field):
        """Validate collection ID is selected and exists in 'collections' table."""
        if field.data == -1:
            raise ValidationError(_('A collection must be selected.'))
        if not Collection.get_by_id(field.data):
            raise ValidationError(_('Collection ID "%(collection_id)s" does not exist',
                                    collection_id=field.data))
```

Reply:
Ah, good catch!